pkg/controllers/apply_controller.go

```diff
@@ -365,17 +379,11 @@ func findObservedGenerationOfManifest(
 // setSpecHashAnnotation computes the hash of the provided spec and sets an annotation of the
 // hash on the provided unstructured objectt. This method is used internally by Apply<type> methods.
 func setSpecHashAnnotation(obj *unstructured.Unstructured) error {
-	data := obj.DeepCopy().Object
-	// do not hash metadata and status section
-	delete(data, "metadata")
-	delete(data, "status")
-
-	jsonBytes, err := json.Marshal(data)
+	specHash, err := generateSpecHash(obj)
```

Comment:
Why do we need to have this function when it's only being used once?

Reply:
S in SOLID, that is every module, class, function should have a single responsibility. The routine is to set the hash annotation, not calculate the hash. Thus the calculation is the responsibility of another routine.